appinventor/components/src/com/google/appinventor/components/runtime/ListView.java

```diff
@@ -153,9 +153,8 @@ public ListView(ComponentContainer container) {
       @Override
       public void onTextChanged(CharSequence cs, int arg1, int arg2, int arg3) {
         // When user changed the Text
-        if (cs.length() == 0) {
-          setAdapterData();
-        } else {
+        setAdapterData();
```

Comment:
I don't think we ever need to call this from inside the filtering function as the list view data won't change during filtering. The main point of the recycler view is that it reuses view objects to prevent having to create lots of memory management overhead. By calling this here we're forcing the view to do a lot of additional work.